actions/login-to-gcs/action.yaml

```
@@ -0,0 +1,62 @@
+name: Login to GCS
+description: Composite action to login to Google Cloud Storage
+inputs:
+  bucket:
+    description: |
+      Name of bucket to upload to. Will default to grafanalabs-${repository.name}-${environment}
+    default: ""
+  environment:
+    description: |
+      Environment for uploading objects (can be either dev or prod).
+    default: dev
+
+outputs:
+  bucket:
+    description: "The name of the bucket that we have authenticated to."
+    value: ${{ steps.get-bucket-name.outputs.bucket }}
+
+runs:
+  using: composite
+  steps:
+    - name: Resolve GCP project
+      id: resolve-project
+      shell: bash
+      run: |
+        if [ "${{ inputs.environment }}" = "dev" ]; then
+          PROJECT="grafanalabs-dev"
+        elif [ "${{ inputs.environment }}" == "prod" ]; then
```

nit: noticed the two use different comparison operators

```suggestion
        elif [ "${{ inputs.environment }}" = "prod" ]; then
```

---

If we are using bash which https://github.com/grafana/shared-workflows/pull/40/files/ba125891a150ef68376d045749690d376437b07d#diff-8ad545e598c1ecec7544d085676f0f79de19ebdb2dc8309213d1ae2f8144497eR23 implies, we should use the builtin `[[ <TEST> ]]` form and `==`. Only if we are targeting POSIX sh should we use the test `[` binary and `=` equality comparison.

---

why _should_?

---

The bash builtin is faster, has more features, and more consistent syntax but ultimately in these cases they work the same. 

I think rules like this that encourage consistency also improve readability.

---

okey dokey. no particular objection - just that if we're going to ask people to abide by something as a rule then let's write it down if not enforce by CI.

we could defer to [the google guide](https://google.github.io/styleguide/shellguide.html) which does cover this case. If you want to make this a requirement, could you raise a PR please?

---

Yep, this was a silly bug. Fixed.

@jdbaldry I'm rather agnostic about how people choose to write their CI scripts (if it works it works), but if you'd like to put together recommendations in a PR we'll be happy to review. Maybe as part of the [GHA user docs](https://github.com/grafana/deployment_tools/blob/master/docs/platform/continuous-integration/github-actions-user-docs.md)?

---

Yeah, I don't have a problem with "if it works, it works". My drive-by comment was mostly to share knowledge about preferred practice rather than required.

That being said, given the esoteric syntaxes and number of quirks of shell scripting, I think you can get a lot from using tooling to lint and teach. I think shellcheck does a really good job of helping you learn how the shell is working so that you don't think something is working when it isn't.

---

@ricky-undeadcoders  If we're using `==` as recommended, use `[[` and `]]` too to use `bash`'s version of `test`.